utilities/utility.py

```diff
@@ -14,6 +16,30 @@ def update_entity_fields(entity, **kwargs):
     return entity
 
 
+def cascade_soft_delete(parent_model, child, parent_id):
+    @event.listens_for(parent_model, 'after_update')
+    def receive_after_update(mapper, connection, target):
+        @event.listens_for(db_session, "after_flush", once=True)
+        def receive_after_flush(session, context):
+            if target.state == 'archived':
+                child_model = getattr(api, child)
+                # Check for resource name is due to the improper
+                # naming of folder room_resource and class Resource
+                child_object = getattr(
+                    child_model.models,
+                    'Resource' if child.capitalize() == 'Room_resource' else child.capitalize()  # noqa: E501
+                )
+                parent_s_id = getattr(child_object, parent_id)
```

Comment:
I think parents_id would be more intuitive the s seems misplaced in the above name.

Reply:
parents_id would refer to more than one parent. 
parent_s_id replaces ' in parent's hence its use here.